### ISSUE TITLE
[STORM-2122] Supervisor V2 can use a lot more memory

### ISSUE THREAD

**Issue by jira-importer** [apache/storm#5904]
<p>When launching a worker the supervisor will read in the complete topology to get a small amount of data out of it.  This can be very large.  In the past we would launch the workers one at a time, so we only needed enough memory to read it in once, now we are doing it in parallel, and lots of these tend to show up very close to one another.  We need to find a way to make it so that we read the data one, and ideally cache it someplace that everyone can share.</p>

---
<details><summary><i>Originally reported by <a title="Robert Joseph Evans" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=revans2">revans2</a>, imported from: <a href="https://issues.apache.org/jira/browse/STORM-2122" target="_blank">Supervisor V2 can use a lot more memory</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Robert Joseph Evans" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=revans2">revans2</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-09-25T13:11:04+01:00
<li><b>imported</b>: 2025-01-24
</ul></i>
</details>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kabhwan">kabhwan</a>:</i>
<p>Thanks <a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=revans2" class="user-hover" rel="revans2">Robert Joseph Evans</a>, I merged into master.</p>
